src/services/blocks/trace/Trace.ts

```diff
@@ -0,0 +1,690 @@
+		//								^^^^^^^^^^^^^^^^^^^^storage item prefix
+		// Since this is a transparent key with the address at the end we can pull out
+		// the address from the key. Here we slice off the storage key + account hash
+		// (which is equal to 96 bytes).
+		const addressRaw = event.data.stringValues.key?.slice(96);
+		if (!(typeof addressRaw === 'string')) {
+			throw new InternalServerError(
+				'Expect encoded address in system::account event storage key'
+			);
+		}
+		const address = this.registry.createType('Address', `0x${addressRaw}`);
+
+		const { value_encoded, result_encoded } = event.data.stringValues;
+		const accountInfoEncoded = value_encoded ?? result_encoded;
+		if (!(typeof accountInfoEncoded === 'string')) {
+			throw new InternalServerError(
+				'Expect accountInfoEncoded to always be a string in system::Account event'
+			);
+		}
+		const accountInfo = (this.registry.createType(
+			'AccountInfo',
+			`0x${accountInfoEncoded.slice(2)}` // Slice off the leading Option byte
+		) as unknown) as AccountInfo;
+
+		return { ...event, accountInfo, address };
+	}
+
+	/**
+	 * Derive `Operation`s based on consecutive Get/Put events to an addresses `accountInfo`
+	 * storage item.
+	 *
+	 * `Operation`s are created by checking for deltas in each field of `accountInfo` based
+	 * on the previous event vs the current event.
+	 *
+	 * @param accountEventsByAddress map of address => events of that addresses `accountInfo`
+	 * events.
+	 */
+	private deriveOperations(
+		accountEventsByAddress: AccountEventsByAddress
+	): Operation[] {
+		const ops = [];
+		for (const events of accountEventsByAddress.values()) {
+			// Note: we sort the events in order so we know the deltas are correct between
+			// previous event <=> current event.
+			const sortedEvents = events.sort((a, b) => a.eventIndex - b.eventIndex);
+			for (const [i, e] of sortedEvents.entries()) {
+				if (i === 0) {
+					// Skip the first event; we always compare previous event <=> current event.
+					continue;
+				}
+
+				// Previous accountInfo.data
+				const prev = events[i - 1].accountInfo.data;
+				// Current accountInfo.data
+				const cur = e.accountInfo.data;
+
+				const free = cur.free.sub(prev.free);
+				const reserved = cur.reserved.sub(prev.reserved);
+				const miscFrozen = cur.miscFrozen.sub(prev.miscFrozen);
+				const feeFrozen = cur.feeFrozen.sub(prev.feeFrozen);
+
+				// Information that will go into any operation we create.
+				const baseInfo = {
+					phase: e.phase,
+					parentSpanId: e.parentSpanId,
+					primarySpanId: e.primarySpanId,
+					eventIndex: e.eventIndex,
+					address: e.address,
+				};
+				const currency = {
+					// For now we assume everything is always in the same token
+					symbol: this.registry.chainTokens[0],
+				};
+				// Base information for all `storage` values
+				const systemAccountData = {
+					pallet: 'system',
+					item: 'Account',
+					field1: 'data',
+				};
+
+				if (!free.eqn(0)) {
```

Comment:
we just filter out balance operations that doesn't have any impact on the state (i.e, 0)?

Reply:
Yeah, basically if nothing changes it probably meant there where back to back reads (normal can happen in block set up and then again prior to extrinsic execution). But yes in cases where there is a put and get we effectively don't count it. I should maybe make a note of that somewhere because it could affect auditing.